### PR TITLE
trafficserver 9.0.1

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Also, depend on `yaml-cpp`. Otherwise, it builds its own.

See also #72714, #72717.

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks -- yep, saw that, and read the suggestion to pass `-undefined dynamic_lookup` to the linker, which is what I tried here.

However, there seems to be another problem of a missing header.

**Comment by mitchblank** [comment]
Hmm wonder if it is trying to use https://github.com/apache/trafficserver/blob/master/tools/http_load/port.h but having some include-path issue?

**Comment by bryancall** [comment]
The `port.h` file that it is looking is the OS file not the one in the http_load directory.  When running configure it is detecting the function `port_create` and trying to use Solaris ports.  In the configure output; `configure: Using port event interface`.  I might remove that check or reorder the checks, since we no longer support Solaris.

**Comment by carlocab** [comment]
Thanks, @bryancall.

If you get around to doing that, I can try applying the patch here to see if it helps. A new release tag would be ideal, but I don't want to impose on your release schedule.

**Comment by bryancall** [comment]
I created a PR on our master branch that should hopefully fix this issue: https://github.com/apache/trafficserver/pull/7597

@zwoop will cherry pick the issue into the 9.0.x branch when it has been merged.

**Comment by bryancall** [comment]
Is it possible to track down where `port_create` is being defined?  My guess is the function is defined somewhere under `/usr/local/include`.

**Comment by carlocab** [comment]
I `grep`ped my `/usr/local/include` (and also `/usr/local/opt/tcl-tk`, since that's not in `/usr/local`) after installing all the dependencies, but that turned up nothing...

**Comment by mitchblank** [comment]
It looks to me like the function detection is just mis-detecting *every* function as existing:
```
checking for inotify_init... yes
checking for port_create... yes
```
investigating...

**Comment by mitchblank** [comment]
@carlocab -- the issue is that since `-undefined dynamic_lookup` is getting passed in, so if the conftest.c *doesn't* find the function then it is not reported as an error:
```
configure:25956: checking for inotify_init
configure:25956: clang -o conftest  -D_GNU_SOURCE -I/usr/local/opt/openssl@1.1/include -DOPENSSL_NO_SSL_INTERN -I/usr/local/include -undefined dynamic_lookup -L/usr/local/opt/openssl
@1.1/lib -L/usr/local/lib conftest.c  >&5
configure:25956: $? = 0
configure:25956: result: yes
configure:25968: checking for port_create
configure:25968: clang -o conftest  -D_GNU_SOURCE -I/usr/local/opt/openssl@1.1/include -DOPENSSL_NO_SSL_INTERN -I/usr/local/include -undefined dynamic_lookup -L/usr/local/opt/openssl
@1.1/lib -L/usr/local/lib conftest.c  >&5
configure:25968: $? = 0
configure:25968: result: yes
```

**Comment by carlocab** [comment]
Aha. Good catch. I guess we can move that call to after `configure`?

File: Formula/trafficserver.rb [hunk 1]
```diff
@@ -41,7 +43,8 @@ def install
       --localstatedir=#{var}
       --sysconfdir=#{etc}/trafficserver
       --with-openssl=#{Formula["openssl@1.1"].opt_prefix}
-      --with-tcl=#{MacOS.sdk_path}/System/Library/Frameworks/Tcl.framework
+      --with-tcl=#{Formula["tcl-tk"].opt_lib}
```
Comment (masaori335) [inline Formula/trafficserver.rb:46]: FWIW, ATS has removed TCL dependency since 9.0.0. 
```
configure: WARNING: unrecognized options: --with-tcl
```

Comment (carlocab) [reply Formula/trafficserver.rb]: Thanks; I removed the `tcl-tk` dependency.

**Comment by carlocab** [comment]
In progress, @BrewTestBot.

**Comment by carlocab** [comment]
ARM build failure:
```
Undefined symbols for architecture arm64:
  "thread-local wrapper routine for pluginThreadContext", referenced from:
      _TSRemapInit in plugin_v2_la-plugin_misc_cb.o
      _TSRemapNewInstance in plugin_v2_la-plugin_misc_cb.o
ld: symbol(s) not found for architecture arm64
```
Looks like the same one as last time.

**Comment by carlocab** [comment]
This doesn't have a Big Sur bottle, so the Big Sur failure shouldn't be blocking. Merging this if this builds on Mojave and Catalina.

**Pull request closed without merging**